### PR TITLE
bugfix 1a2b game

### PR DESCRIPTION
**Pull request by wow890209**
## Why need this change? / Root cause: 
-新增不是文字頻道的檢查，防止在語音頻道、討論串嘗試建立遊戲
## Changes made:
-
## Test Scope / Change impact:
-

### CODE REVIEW

File: guess-num-1A2B/src/main/kotlin/tw/waterballsa/utopia/guessnum1a2b/GuessNum1A2BListener.kt [hunk 1]
```diff
@@ -43,7 +44,12 @@ class GuessNum1A2BListener : UtopiaListener() {
     }
 
     private fun SlashCommandInteractionEvent.isValidCommandName(): Boolean {
-        return fullCommandName == GUESS_NUM_1A2B_COMMAND_NAME
+        var isVaild = fullCommandName == GUESS_NUM_1A2B_COMMAND_NAME
+        if (isVaild && channel is TextChannel) {
```
Comment (Wally5077) [inline guess-num-1A2B/src/main/kotlin/tw/waterballsa/utopia/guessnum1a2b/GuessNum1A2BListener.kt:48]: `channel is TextChannel` 並不屬於 isValidCommandName 的邏輯，你應該放在方法外面

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done.

File: guess-num-1A2B/src/main/kotlin/tw/waterballsa/utopia/guessnum1a2b/GuessNum1A2BListener.kt [hunk 2]
```diff
@@ -68,8 +73,11 @@ class GuessNum1A2BListener : UtopiaListener() {
         }
     }
 
-    private fun MessageReceivedEvent.getRoom(): ThreadChannel {
-        return channel.asThreadChannel()
+    private fun MessageReceivedEvent.getRoom(): ThreadChannel? {
+        if (channel is ThreadChannel) {
+            return channel.asThreadChannel()
+        }
+        return null
```
Comment (Wally5077) [inline guess-num-1A2B/src/main/kotlin/tw/waterballsa/utopia/guessnum1a2b/GuessNum1A2BListener.kt:80]: `if channel is ThreadChannel channel.asThreadChannel() else null`

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done.

**Review by Wally5077 — APPROVED** [review]
LGTM

**Review by frankvicky — APPROVED** [review]

**Pull request merged**